num-format/src/system_locale/windows.rs

```diff
@@ -24,18 +25,13 @@ use crate::strings::{DecString, InfString, MinString, NanString, PlusString, Sep
 use crate::system_locale::SystemLocale;
 
 lazy_static! {
-    static ref SYSTEM_DEFAULT: Result<&'static str, Error> = {
-        CStr::from_bytes_with_nul(LOCALE_NAME_SYSTEM_DEFAULT).map_err(|_| {
+    static ref SYSTEM_DEFAULT: Result<String, Error> = {
+        Ok(unsafe{U16CStr::from_ptr_str(LOCALE_NAME_SYSTEM_DEFAULT)}.to_string().map_err(|_| {
```

Comment:
Would you mind explaining this change? It's not quite intuitive why the old code doesn't work any more when switching to windows-sys.